### PR TITLE
update to v2

### PR DESCRIPTION
**Pull request by guolinke**
updates:

1. A new sampling algorithm- Gradient-based One-Side Sampling(GOSS). Can learn from subset while achieving almost the same accuracy. 
2. Exclusive Feature Bundling(EFB). The upgrade version of categorical feature support. EFB can significantly speed up the training from large sparse data. Since Exclusive Feature Bundling can efficiently  handle one-hot coding features as well, current version of categorical feature support will be removed. 

The related paper will release soon as well.

todo:

- [x]  remove categorical features support in doc/python-package/r-package

### CODE REVIEW

**Comment by guolinke** [comment]
@wxchan can you help to check the the removing of categorical feature in python package?

**Comment by wxchan** [comment]
Does it mean users need to do one-hot encoding themselves now?

File: tests/python_package_test/test_basic.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def test(self):
         for preds in zip(pred_from_matr, pred_from_model_file):
             self.assertEqual(*preds)
         # check pmml
-        os.system('python ../../pmml/pmml.py model.txt')
+        # os.system('python ../../pmml/pmml.py model.txt')
```
Comment (wxchan) [inline tests/python_package_test/test_basic.py]: why comments out this line?

**Comment by guolinke** [comment]
@wxchan yes. LightGBM now can handle one-hot features efficiently. It provides the same training speed as categorical features.

File: docs/Python-API.md [hunk 2]
```diff
@@ -5,8 +5,8 @@
     - [Booster](Python-API.md#booster)
 
 * [Training API](Python-API.md#training-api)
-    - [train](Python-API.md#trainparams-train_set-num_boost_round100-valid_setsnone-valid_namesnone-fobjnone-fevalnone-init_modelnone-feature_nameauto-categorical_featureauto-early_stopping_roundsnone-evals_resultnone-verbose_evaltrue-learning_ratesnone-callbacksnone)
-    - [cv](Python-API.md#cvparams-train_set-num_boost_round10-data_splitternone-nfold5-stratifiedfalse-shuffletrue-metricsnone-fobjnone-fevalnone-init_modelnone-feature_nameauto-categorical_featureauto-early_stopping_roundsnone-fpreprocnone-verbose_evalnone-show_stdvtrue-seed0-callbacksnone)
```
Comment (wxchan) [inline docs/Python-API.md]: `data_splitternone-` should not be removed here

File: docs/Python-API.md [hunk 3]
```diff
@@ -536,7 +515,7 @@ The methods of each Class is in alphabetical order.
     booster : a trained booster model
 
 
-####cv(params, train_set, num_boost_round=10, data_splitter=None, nfold=5, stratified=False, shuffle=True, metrics=None, fobj=None, feval=None, init_model=None, feature_name='auto', categorical_feature='auto', early_stopping_rounds=None, fpreproc=None, verbose_eval=None, show_stdv=True, seed=0, callbacks=None)
```
Comment (wxchan) [inline docs/Python-API.md]: same here, `data_splitternone-`

File: tests/python_package_test/test_basic.py (hunk 1 above)
Comment (wxchan) [inline tests/python_package_test/test_basic.py]: Should we still keep pmml directory? I think https://github.com/jpmml/jpmml-lightgbm does a better job.

**Comment by guolinke** [comment]
@wxchan i forget to fix these. Can you create a new PR to fix ?
